robot-server/robot_server/app.py

```diff
@@ -59,19 +54,14 @@
 async def on_startup() -> None:
     """Handle app startup."""
     initialize_logging()
-    # Initialize api
-    (await get_hardware_wrapper()).async_initialize()
+    initialize_hardware(app.state)
 
 
 @app.on_event("shutdown")
 async def on_shutdown() -> None:
     """Handle app shutdown."""
-    s = await get_rpc_server()
-    await s.on_shutdown()
-    # Remove all sessions
-    await (await get_session_manager()).remove_all()
-    # Remove all uploaded protocols
-    (await get_protocol_manager()).remove_all()
+    await cleanup_rpc_server(app.state)
+    await cleanup_hardware(app.state)
```

Comment:
`cleanup_rpc_server` may still raise due to #8155. I wonder if this makes more sense with a gather and some logs:

```suggestion
    shutdown_results = await asyncio.gather(
        cleanup_rpc_server(app.state),
        cleanup_hardware(app.state),
        return_exceptions=True,
    )
    
    shutdown_errors = [r for r in shutdown_results if isinstance(r, Exception)]
    
    for e in shutdown_errors:
        log.warning(f"Error encountered during shutdown", exc_info=e) 
```